packages/shared/lib/services/sync/data/data.service.ts

```diff
@@ -33,11 +33,12 @@ export async function upsert(
 
     const addedKeys = await getAddedKeys(recordsWithoutDuplicates, nangoConnectionId, model);
     const updatedKeys = await getUpdatedKeys(recordsWithoutDuplicates, nangoConnectionId, model);
+    const deletedKeys = await getDeletedKeys(recordsWithoutDuplicates, nangoConnectionId, model);
```

Comment:
for legacy records I went the easy route, creating a new function. Given this code is going away very soon I didn't feel like refactoring things to avoid making an extra query.